### DISCUSSION TITLE
fpart.log vs fpsync.log

### DISCUSSION THREAD

**Discussion by jay-law** [martymac/fpart#63] [General]
Hello Ganael,

Fpart and Fpsync are amazing tools, thank you so much for creating them!  

Can you provide insight into when `fpart.log` gets created opposed to `fpsync.log`?  From what I can tell, only one gets created under `/tmp/fpsync/log/<run_id>/` but I'm unable to identify which condition/flag/options determines the "main" log file.

Additionally, when do empty logs get removed?   Some of my executions have a ton of empty log files while other runs have no empty log files.

Running version 1.6.1 on Rocky 9.

Example execution:
```
sudo fpsync -n 20 -vvvvv -O "-X|one|-X|two|-X|three|" \
    -o "-nl --out-format=\\\"%i||%M||%n||%L\\\"" \
    /source_dir/ \
    /dest_dir/
```

**Comment by martymac** [comment]
Hello @jay-law,

v1.6.1 is "next" (devel) version, please be aware that you may encounter some bugs. There have been several changes regarding logging since 1.6.0 (last release), see Changelog.md.

> Can you provide insight into when fpart.log gets created opposed to fpsync.log? From what I can tell, only one gets created under /tmp/fpsync/log/<run_id>/ but I'm unable to identify which condition/flag/options determines the "main" log file.

In 1.6.0 and previous versions, main log file was: /tmp/fpsync/log/<run_id>/fpart.log. For consistency, it has been renamed to /tmp/fpsync/log/<run_id>/fpsync.log as that file contains logs for fpsync's <run_id> run (and not only fpart logs). This is the (only) main log file for that run.

> Additionally, when do empty logs get removed? Some of my executions have a ton of empty log files while other runs have no empty log files.

They get removed when a job is considered as finished, see the end of work_list_refresh() function. Do you still have empty log files with an up-to-date version of the repository ?

**Comment by jay-law** [comment]
Hello @martymac,

I appreciate you responding.  The `fpsync` version installed was not consistent across all users.  After fixing this issue, everything is working as expected.  

Thanks again for creating `fpart` and `fpsync`!
